typespec-extension/src/code-model-builder.ts

```diff
@@ -2520,12 +2522,14 @@ export class CodeModelBuilder {
   }
 
   private getSerializedName(target: ModelProperty): string {
-    // First get projected name, if not found, return target.name
+    // First get projected name, if not found, call resolveEncodedName
+    // TODO: deprecate getProjectedName
     const jsonProjectedName = getProjectedName(this.program, target, "json");
     if (jsonProjectedName) {
       return jsonProjectedName;
     }
-    return target.name;
+    // TODO: the MIME better be from SchemaUsage.serializationFormats. However, at this step, the values is not propagated to inner models.
+    return resolveEncodedName(this.program, target, "application/json");
```

Comment:
I had to put it after `projectedName`, as this `resolveEncodedName` won't tell us whether it resolved it from `@encodedName`, or fallback to target.name.